research/conditional/utils/argparse.py

```diff
@@ -25,6 +25,30 @@ def introduce_parser_arguments(
     parser.add_argument("--no_ff", action="store_true")
     parser.add_argument("--moe_inner_expert", type=str, default="ff")
 
+    parser.add_argument(
+        "--double_routing_from_start",
+        action="store_true",
+        help="Whether second routing is calculated before 1 linear",
+    )
+    parser.add_argument(
+        "--double_routing_use_single",
+        action="store_true",
+        help="Mock single routing with double routing",
+    )
+    parser.add_argument(
+        "--double_mot", action="store_true", help="Double routing with MoT"
+    )
+    parser.add_argument(
+        "--double_routing_before_relu",
+        action="store_true",
+        help="Whether second routing is calculated before relu",
+    )
+    parser.add_argument(
+        "--double_routing_snd_ln",
+        action="store_true",
+        help="Whether second routing has another layer norm",
+    )
+
```

Comment:
I'd shorten `double_routing` to `dr` and use this save in length to write more descriptive option names, like for example `double_routing_use_single` could be named `dr_mock_single` (although I still don't know what mocking single routing with double routing means)

Reply:
Ok, I've shortened or removed them (some were obsolete and usable only on previous versions)